File: problems/3614-process-string-with-special-operations-ii/solution.py
class Solution:
    def process_string(s, k):
        result = ""
        for char in s:
            if 'a' <= char <= 'z':
                result += char
            elif char == '*':
                if result:
                    result = result[:-1]
            elif char == '#':
                result += result
            elif char == '%':
                result = result[::-1]

        if k >= len(result) or k < 0:
            return "."
        else:
            return result[k]

File: problems/3614-process-string-with-special-operations-ii/solution.cpp
class Solution {
public:
    char processStr(string s, long long k) {
    #include <string>
    #include <algorithm>

    char solve(std::string s, long long k) {
        std::string result = "";
        for (char c : s) {
            if (c >= 'a' && c <= 'z') {
                result += c;
            } else if (c == '*') {
                if (!result.empty()) {
                    result.pop_back();
                }
            } else if (c == '#') {
                result += result;
            } else if (c == '%') {
                std::reverse(result.begin(), result.end());
            }
        }
        if (k >= result.length() || k < 0) {
            return '.';
        }
        return result[k];
    }
};